zkevm-circuits/src/evm_circuit/witness.rs

```diff
@@ -331,56 +332,41 @@ impl Bytecode {
         Self { hash, bytes }
     }
 
-    pub fn table_assignments<'a, F: FieldExt>(
-        &'a self,
-        randomness: F,
-    ) -> impl Iterator<Item = [F; 4]> + '_ {
-        struct BytecodeIterator<'a, F> {
-            idx: usize,
-            push_data_left: usize,
-            hash: F,
-            bytes: &'a [u8],
-        }
-
-        impl<'a, F: FieldExt> Iterator for BytecodeIterator<'a, F> {
-            type Item = [F; 4];
-
-            fn next(&mut self) -> Option<Self::Item> {
-                if self.idx == self.bytes.len() {
-                    return None;
-                }
-
-                let idx = self.idx;
-                let byte = self.bytes[self.idx];
+    pub fn table_assignments<F: FieldExt>(&self, randomness: F) -> Vec<[F; 5]> {
+        let n = 1 + self.bytes.len();
+        let mut rows = Vec::with_capacity(n);
+        let hash =
+            RandomLinearCombination::random_linear_combine(self.hash.to_le_bytes(), randomness);
+        let mut push_data_left = 0;
+        for idx in 0..n {
+            if idx == 0 {
+                rows.push([
```

Comment:
move this part out of the for loop and you don't need to shift the index like `let i = idx - 1;`.